### ISSUE TITLE
Missing parenthesis on lint

### ISSUE THREAD

**Issue by ph04** [rust-lang/rust-clippy#9362]
### Summary

The lint is wrong because it's missing necessary parenthesis.

### Reproducer

I tried this code:

```rust
fn main() {
    let first: [u8; 10] = [0; 10];
    let second: [u8; 10] = [0; 10];

    let _: usize = first
        .iter()
        .zip(second.iter())
        .map(|(f, s)| (*f as isize - *s as isize).abs() as usize)
        .sum();
}
```
and I got this lint

```
warning: casting the result of `isize::abs()` to usize
 --> src/main.rs:8:23
  |
8 |         .map(|(f, s)| (*f as isize - *s as isize).abs() as usize)
  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `*f as isize - *s as isize.unsigned_abs()`
  |
  = note: `#[warn(clippy::cast_abs_to_unsigned)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_abs_to_unsigned
```

which is wrong because it should be `(*f as isize - *s as isize).unsigned_abs()` instead.

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by giraffate** [comment]
Thanks for the report!

This has been fixed by https://github.com/rust-lang/rust-clippy/pull/9266, so I'm closing this.
